src/compas/datastructures/mesh/mesh_to_points.py

```diff
@@ -0,0 +1,93 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+from numpy import array
+from numpy.random import choice
+from numpy.random import rand
+from numpy import sqrt
+
+
+
+
+__all__ = [
+    'mesh_to_points',
+]
+
+def mesh_to_points(mesh, num_points: int = 10000):
+    """Convert a mesh to points
+
+    Parameters
+    ----------
+    mesh : compas.datastructures.Mesh
+        The mesh data structure.
+    num_points : (int) 
+        How many points sampled
+    return_normals : (bool)  (TODO)
+        Return normals for the points, if True
+
+    Returns
+    -------
+    list
+        A numpy array with dim = [num_points, 3].
+
+    Examples
+    --------
+
+
+    """
+    if mesh.is_empty():
+    raise ValueError("Mesh is empty.")
+    if mesh.is_quadmesh():
+        mesh.quads_to_triangles()
+    if not mesh.is_valid():
+        raise ValueError("Mesh is invalid.")
+        
+    # (1)  acquire the list of area
+    first_idx = list(mesh.face.keys())[0]
+    add_idx = 0
+    area_list=[]
+    if first_idx != 0:
+        add_idx = first_idx
+    for i in range(len(mesh.face)):
+        area_list.append(mesh.face_area(i+add_idx))
+    area_list = array(area_list)
+    area_list_norm = area_list / area_list.sum()
+
+    # (2) sample num_points on a mesh face regarding its weight of area
+    mesh_face_idx = list(mesh.face.keys())
+    sample_face_idxs = choice(mesh_face_idx, num_points, p=area_list_norm) - add_idx
```

Comment:
i am assuming you created `add_idx` to make sure face identifiers map properly to array indices.
the way you are doing this now only takes care of the case when the first face identifier is not `0` assuming that from then onward the indices form a continuous and contiguous series. this is not necessarily the case. the proper way to do this would be to use a `face_index` map

```python
face_index = {face: index for index, face in enumerate(mesh.faces())}
faces = list(mesh.faces())  # unless you already did this above
sample_faces = [face_index[index] for face in choice(faces, num_points, p=area_list_norm)]
```
or 
```python
faces = list(range(mesh.number_of_faces()))
samples_faces = choice(faces, num_points, p=area_list_norm)
```

Reply:
there is a new tutorial article related to this that might clarify things
https://compas.dev/compas/latest/tutorial/numericaldata.html